### PR TITLE
Added a simple ping counter

### PR DESCRIPTION
**Pull request by halcyo-n**
Simple ping counter that renders the same way as CG_DrawFPS
I'm sure there are better ways to do it but as commented in the code i don't want to mess things up, this is simply a start but if you're aiming for a early pre release this month i'm sure the playerbase would appreciate a ping counter.

### CODE REVIEW

File: src/client/game/dvars.cpp [hunk 1]
```diff
@@ -18,6 +18,7 @@ namespace dvars
 	game::dvar_t* g_speed = nullptr;
 
 	game::dvar_t* pm_bouncing = nullptr;
+    game::dvar_t* cg_drawPing = nullptr;
```
Comment (momo5502) [inline src/client/game/dvars.cpp:21]: Formatting...

File: src/client/module/fps.cpp [hunk 2]
```diff
@@ -14,6 +14,7 @@ namespace fps
 	{
 		float fps_color[4] = {0.6f, 1.0f, 0.0f, 1.0f};
 		float origin_color[4] = {1.0f, 0.67f, 0.13f, 1.0f};
+		float ping_color[4] = { 1.0f, 1.0f, 1.0f, 0.65f };
```
Comment (momo5502) [inline src/client/module/fps.cpp:17]: Formatting...

File: src/client/module/fps.cpp [hunk 3]
```diff
@@ -121,11 +122,43 @@ namespace fps
 			}
 		}
 
+		void cg_draw_ping()
+		{
+			const auto* draw_ping = game::Dvar_FindVar("cg_drawPing");
+			if (draw_ping && draw_ping->current.integer != 0 && game::CL_IsCgameInitialized())
+			{
+				//garbage c code btw -halcyon
+				uintptr_t ping_offset = 0x1419E5100;
```
Comment (momo5502) [inline src/client/module/fps.cpp:131]: What's the point of saving that value? Just directly do a reinterpret cast

File: src/client/module/fps.cpp [hunk 4]
```diff
@@ -121,11 +122,43 @@ namespace fps
...
+
+				auto* const ping_string = utils::string::va("Ping: %i", ping);
+
+				const auto scale = 1.0f;
+
+				const auto x = (game::ScrPlace_GetViewPlacement()->realViewportSize[0] - 50.0f) - game::R_TextWidth(
+				ping_string, 0x7FFFFFFF, font) * scale;
+
+				const auto y = font->pixelHeight * 1.2f;
+				//todo: implement a check if drawFPS is >1, then push ping counter under fps counter
```
Comment (momo5502) [inline src/client/module/fps.cpp:145]: Why are you creating a pull request if your code is unfinished?
Directly place the ping counter at a suited position, don't adapt the behaviour depending on cg_drawfps

File: src/client/module/fps.cpp [hunk 5]
```diff
@@ -121,11 +122,43 @@ namespace fps
...
+				//todo: implement this into scoreboard? i don't wanna touch the scoreboard considering the state it's in at the moment.
```
Comment (momo5502) [inline src/client/module/fps.cpp:146]: Yes, this would actually be the only reasonable solution

File: src/client/module/fps.cpp [hunk 6]
```diff
@@ -145,7 +178,11 @@ namespace fps
 			// change cg_drawfps flags to saved
 			utils::hook::call(SELECT_VALUE(0x1401F400A, 0x140272B98), &cg_draw_fps_register_stub);
 
+			// change cg_drawping flags to saved
+			//utils::hook::call(SELECT_VALUE(0x1419E50FF, 0x1419E5100), &cg_draw_ping_register_stub);
```
Comment (momo5502) [inline src/client/module/fps.cpp:182]: Why?

File: src/client/module/fps.cpp [hunk 7]
```diff
@@ -121,11 +122,43 @@ namespace fps
...
+				game::R_AddCmdDrawText(ping_string, 0x7FFFFFFF, font, x, y, scale, scale, 0.0f, ping_color, 6);
+			}
+		}
+
 		void cg_draw_fps_register_stub(const char* name, const char** _enum, const int value, unsigned int flags,
 		                               const char* desc)
 		{
 			game::Dvar_RegisterEnum(name, _enum, value, 0x1, desc);
 		}
+		void cg_draw_ping_register_stub(const char* name, const char** _enum, const int value, unsigned int flags,
```
Comment (momo5502) [inline src/client/module/fps.cpp:157]: Empty line?

**Review by momo5502 — CHANGES_REQUESTED** [review]
Please, don't commit submodule changes. Merge the master into your fork if you didn't touch the submodules.
The whole approach is kinda meh, changing the scoreboard would be way better, but I guess it's fine until there is a way to adjust the scoreboard

**Comment by halcyo-n** [comment]
Had a hard time working with the formatting changes cus github desktop is stupid, hopefully all is well now

File: src/client/game/dvars.cpp (hunk 1 above)
Comment (momo5502) [reply src/client/game/dvars.cpp:21]: formatting is still fucked.

File: src/client/module/fps.cpp (hunk 2 above)
Comment (momo5502) [reply src/client/module/fps.cpp:17]: formatting is still fucked?

**Review by momo5502 — APPROVED** [review]
cba to wait for you to fix the formatting, I'll fix it myself

**Comment by halcyo-n** [comment]
Formatting was aligned within VS but github desktop didn't seem to pick up on the changes

**Comment by momo5502** [comment]
You also committed a submodule and your dvar registration was completely separate from your usage. 
I created a commit that fixes those things: https://github.com/XLabsProject/iw6x-client/commit/3fc730313e6a4b5acfc7d644ee25f774d633b10a
Next time, please try to fix these things yourself.

**Pull request merged**